### DISCUSSION TITLE
Why does famous llama.cpp not support GPU fine-tuning?@GG

### DISCUSSION THREAD

**Discussion by jygmysoul** [ggml-org/llama.cpp#8153] [Q&A]
Why does llama.cpp not support GPU fine-tuning?
Using CPU to fine-tune shakespeare.txt of only 100K is very slow. My computer CPU fan is humming. This is very power-consuming and unreasonable. Doesn't GG have a plan to improve this? When will it support GPU fine-tuning? If it supports GPU fine-tuning, what are the requirements? Please support GPU fine-tuning as soon as possible. It is unreasonable to use CPU.

**Comment by jygmysoul** [comment]
Or are there any other alternative solutions or open source projects? The CPU is too slow.
Thank you for your support and dedication!

**Comment by jygmysoul** [comment]
11

**Comment by iplayfast** [comment]
If you compile with gpu enabled it will use the gpu. Go through the readme carefully.
